includes/class-wgpb-products-controller.php

```diff
@@ -230,10 +230,13 @@ public function prepare_item_for_response( $post, $request ) {
 	protected function prepare_objects_query( $request ) {
 		$args = parent::prepare_objects_query( $request );
 
-		$orderby       = $request->get_param( 'orderby' );
-		$order         = $request->get_param( 'order' );
-		$cat_operator  = $request->get_param( 'cat_operator' );
-		$attr_operator = $request->get_param( 'attr_operator' );
+		$orderby            = $request->get_param( 'orderby' );
+		$order              = $request->get_param( 'order' );
+		$cat_operator       = $request->get_param( 'cat_operator' );
+		$attr_operator      = $request->get_param( 'attr_operator' );
+		$attributes         = $request->get_param( 'attributes' );
+		$tax_relation       = $request->get_param( 'tax_relation' );
+		$catalog_visibility = $request->get_param( 'catalog_visibility' );
```

Comment:
I think we'll need to add an item to `get_collection_params()` for this param:

```php
		$params['catalog_visibility'] = array(
			'description'       => __( 'Determines if hidden or visible catalog products are shown.', 'woo-gutenberg-products-block' ),
			'type'              => 'string',
			'enum'              => array( 'visible', 'hidden' ),
			'sanitize_callback' => 'sanitize_text_field',
			'validate_callback' => 'rest_validate_request_arg',
		);
```

Reply:
It would be good to support all visibility options – so the `enum` should be `array( 'visible', 'catalog', 'search', 'hidden' )` – you can see [the property in the item schema here.](https://github.com/woocommerce/woocommerce/blob/9ef6ea0be71a0719d72c5e511eb3c6b61330f183/includes/api/class-wc-rest-products-controller.php#L763-L769)

So `hidden` would get you products in both terms `'exclude-from-search'` & `'exclude-from-catalog'`,  `catalog` would be products `NOT IN 'exclude-from-catalog'`, and `search` would be `NOT IN 'exclude-from-search'`. `visible`, then, is `NOT IN `'exclude-from-search'` or `'exclude-from-catalog'`.

Reply:
Gotcha, thanks! I'll add code and tests.

Reply:
`catalog_visibility` still needs to be added to `get_collection_params` here, the example from Josh above would work well (just update the enum).